src/main/java/com/joss/conductor/mobile/Locomotive.java

```diff
@@ -144,13 +157,17 @@ private void initialize() {
             }
         }
 
-        // Set session ID after driver has been initialized
-        String id = getAppiumDriver().getSessionId().toString();
-        sessionId.set(id);
-
-        // TODO: Added to support biometrics on android until java-client PR #473 is pulled in
-        MobileCommand.commandRepository.put("fingerPrint",
-                MobileCommand.postC("/session/:sessionId/appium/device/finger_print"));
+        // if session could not be established then retry for a limited amount of times
+        if (getAppiumDriver().getSessionId() == null) {
+            if (startCounter < configuration.getStartSessionRetries()) {
```

Comment:
should this be `startCounter <= configuration.getStartSessionRetries()`?  I'm wondering if using the default won't run `1 of 1` since 1 isn't < 1

Reply:
This check is being done at the end of the method after the session is started. So for the case of `1 < 1` the session start has been tried once already.

With that being said I think the readability of this could be improved. Doing a recursive call with a check at the end of the method does not seem ideal. Most text book recursions I have seen have the termination condition at the beginning of the method. I'm not 100% sure on how to do this best and would be very open to opinions/thoughts.